src/Compilers/VisualBasic/Portable/Symbols/MethodSignatureComparer.vb

```diff
@@ -538,8 +543,17 @@ Done:
                                     GoTo Done
                                 End If
                             End If
+                        ElseIf (comparisons And SymbolComparisonResults.TupleNamesMismatch) <> 0 AndAlso
+                            Not type1.Type.IsSameType(type2.Type, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) Then
+
+                            results = results Or SymbolComparisonResults.TupleNamesMismatch
+                            If (stopIfAny And SymbolComparisonResults.TupleNamesMismatch) <> 0 Then
+                                GoTo Done
+                            End If
                         ElseIf (comparisons And SymbolComparisonResults.CustomModifierMismatch) <> 0 AndAlso
-                               (type1 <> type2 OrElse param1.CountOfCustomModifiersPrecedingByRef <> param2.CountOfCustomModifiersPrecedingByRef) Then
+                               (Not type1.IsSameType(type2, TypeCompareKind.IgnoreTupleNames) OrElse
```

Comment:
>TypeCompareKind.IgnoreTupleNames [](start = 60, length = 32)

I think the code will be more robust long term if we wouldn't make an assumption that TypeCompareKind.TupleNamesMismatch is "all - IgnoreCustomModifiersAndArraySizesAndLowerBounds".